packages/dds/tree/src/simple-tree/getJsonSchema.ts

```diff
@@ -0,0 +1,76 @@
+/*!
+ * Copyright (c) Microsoft Corporation and contributors. All rights reserved.
+ * Licensed under the MIT License.
+ */
+
+import type { TreeJsonSchema } from "./jsonSchema.js";
+import type { TreeNodeSchema } from "./schemaTypes.js";
+import { toJsonSchema } from "./simpleSchemaToJsonSchema.js";
+import { getSimpleSchema } from "./getSimpleSchema.js";
+
+/**
+ * Private symbol under which the results of {@link getJsonSchema} are cached on an input {@link TreeNodeSchema}.
+ */
+const jsonSchemaCacheSymbol = Symbol("jsonSchemaCache");
+
+/**
+ * Creates a {@link https://json-schema.org/ | JSON Schema} representation of the provided {@link TreeNodeSchema}.
+ *
+ * @remarks
+ * Useful when communicating the schema to external libraries or services.
+ * Caches the result on the input schema for future calls.
+ *
+ * @example
+ *
+ * A Shared Tree schema like...
+ *
+ * ```typescript
+ * class MyObject extends schemaFactory.object("MyObject", {
+ * 	foo: schemaFactory.number,
+ * 	bar: schemaFactory.optional(schemaFactory.string),
+ * });
+ * ```
+ *
+ * ...will yield JSON Schema like...
+ * ```json
```

Comment:
I believe Markdown best practices call for a line break before and after code blocks.

Reply:
Hmm. Couldn't find anything specifically about fenced code blocks, but there is similar guidance for things like block quotes. And I would think the guidance would be the same.
https://www.markdownguide.org/basic-syntax/#blockquotes-1

Reply:
Not sure if it matters in practice though 🤷‍♂️

Reply:
Except for GitHub because they inexplicably are inconsistent even within their own products, single line breaks in markdown are not significant. Thus, unless you want something interpreted as being on the same line, it should have double line breaks.

Reply:
It
sure
would
be nice
if GitHub followed
CommonMark.